docs/source/howto/run_codes.rst

```diff
@@ -0,0 +1,372 @@
+
+A |Computer| in AiiDA denotes a computational resource on which you will run your calculations.
+It can either be:
+
+ 1. the machine where AiiDA is installed or
+ 2. any machine that is accessible via `SSH <https://en.wikipedia.org/wiki/Secure_Shell>`_ from the machine where AiiDA is installed.
+
+The second option allows managing multiple remote compute resources (including HPC clusters and cloud services) from the same AiiDA installation and moving computational jobs between them.
+
+.. tip::
+
+    The second option requires access through a SSH keypair.
+    If your compute resource demands two-factor authentication, you may need to install AiiDA directly on the compute resource instead.
+
+
+Computer requirements
+---------------------
+
+Requirements for configuring a compute resource in AiiDA are:
+
+* It runs a Unix-like operating system (Linux distros and MacOS should work fine)
+* It has ``bash`` installed
+* (option 2.) It is accessible *via* SSH from the machine that runs AiiDA (possibly :ref:`via a proxy server<how-to:ssh:proxy>`)
+* (optional) It has batch scheduler installed (see the :ref:`list of supported schedulers <topics:schedulers>`)
+
+If you are configuring a remote computer, start by :ref:`configuring password-less SSH access <how-to:ssh>` to it.
+
+.. note::
+
+    AiiDA will use ``bash`` on the remote computer, regardless of the default shell.
+    Please ensure that your remote ``bash`` configuration does not load a different shell.
+
+
+.. _how-to:run-codes:computer:setup:
+
+Computer setup
+--------------
+
+The configuration of computers happens in two steps: setting up the public metadata asociated with the |Computer| in AiiDA provenance graphs, and configuring private connection details.
+
+Start by creating a new computer instance in the database:
+
+.. code-block:: console
+
+   $ verdi computer setup
+
+At the end, the command will open your default editor on a file containing a summary of the configuration up to this point.
+You can add ``bash`` commands that will be executed
+
+ * *before* the actual execution of the job (under 'Pre-execution script'), and
+ * *after* the script submission (under 'Post execution script').
+
+Use these additional lines to perform any further set up of the environment on the computer, for example loading modules or exporting environment variables:
+
+.. code-block:: bash
+
+   export NEWVAR=1
+   source some/file
+
+.. note::
+
+   Don't specify settings here that are specific to a code or calculation: you can set further pre-execution commands at the ``Code`` and even ``CalcJob`` level.
+
+When you are done editing, save and quit (e.g. ``<ESC>:wq<ENTER>`` in ``vim``).
+The computer has now been created in the database but you still need to *configure* access to it using your credentials.
+
+.. tip::
+    In order to avoid having to retype the setup information the next time round, you can provide some (or all) of the information via a configuration file:
```

Comment:
```suggestion
    In order to avoid having to retype the setup information the next time around, you can provide some (or all) of the information via a configuration file:
```